docs/arcaflow/getting-started.md

```diff
@@ -105,142 +189,488 @@ input:
+                  type:
+                    type_id: string
+          root: ErrorOutput
+      success:
+        error: false
+        schema:
+          objects:
+            SuccessOutput:
+              id: SuccessOutput
+              properties:
+                message:
+                  display: {}
+                  required: true
+                  type:
+                    type_id: string
+          root: SuccessOutput
+```
+
+The plugin schema can also be returned in JSON format, in which case you must specify wheter to return either the `input` or `output` schema.
 
 === "Podman"
-    
-    ```yaml title="config.yaml"
-    deployer:
-      type: podman
-      deployment:
-        # Make sure we don't try to pull the image we have locally
-        imagePullPolicy: Never
+    ```bash
+    podman run --rm quay.io/arcalot/arcaflow-plugin-example --json-schema output
+    ```
+=== "Docker"
+    ```bash
+    docker run --rm quay.io/arcalot/arcaflow-plugin-example --json-schema output
     ```
 
-!!! tip
-    Further reading: [Setting up Arcaflow](running/setup.md)
+```json title="example plugin output schema JSON"
+{
+  "$id": "hello-world",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Hello world! outputs",
+  "description": "Says hello :)",
+  "oneof": [
+    {
+      "output_id": {
+        "type": "string",
+        "const": "success"
+      },
+      "output_data": {
+        "type": "object",
+        "properties": {
+          "message": {
+            "type": "string"
+          }
+        },
+        "required": [
+          "message"
+        ],
+        "additionalProperties": false,
+        "dependentRequired": {}
+      }
+    },
+    {
+      "output_id": {
+        "type": "string",
+        "const": "error"
+      },
+      "output_data": {
+        "type": "object",
+        "properties": {
+          "error": {
+            "type": "string"
+          }
+        },
+        "required": [
+          "error"
+        ],
+        "additionalProperties": false,
+        "dependentRequired": {}
+      }
+    }
+  ],
+  "$defs": {
+    "SuccessOutput": {
+      "type": "object",
+      "properties": {
+        "message": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "message"
+      ],
+      "additionalProperties": false,
+      "dependentRequired": {}
+    },
+    "ErrorOutput": {
+      "type": "object",
+      "properties": {
+        "error": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "error"
+      ],
+      "additionalProperties": false,
+      "dependentRequired": {}
+    }
+  }
+}
+```
 
-## Step 7: Run the workflow
+A plugin takes its input as a file, but because it runs as a container, it looks for the input file in the context of the container. This means you either need to bind-mount the input file to the container, or, as in this example, pipe the input value to the plugin's file input.
 
```

Comment:
I think that this text provides the wrong level of detail.  It should either explain _more_ (like how the plugin uses the `-f` to specify the input file on its command line), or it should explain __less__ (like stick with the `stdin` approach and don't mention files or bind-mounts).

And, if you want to go the _more_ route, then out of fairness, you should show how to run the plugin with a bind-mount -- e.g.,
```bash
podman run --rm -v ${PWD}/input.yaml:/input.yaml quay.io/arcalot/arcaflow-plugin-example -f /input.yaml
```
I'm not sure that that is harder to understand than the I/O redirection recommended below....